src/web/lib/components/Onboarding/index.js

```diff
@@ -28,6 +29,15 @@ export default class Onboarding extends React.Component {
 
   handleDismiss = e => {
     if (e.target.classList.contains("dismissable")) {
+      const params = queryString.parse(window.location.search);
+      const pathname = window.location.pathname;
+
+      delete params.firstRun;
+      let stringifiedParams = queryString.stringify(params);
+
+      stringifiedParams = `${stringifiedParams && "?"}${stringifiedParams}`;
+      history.replaceState({}, "", `${pathname}${stringifiedParams}`);
```

Comment:
Use `history.state` instead of `{}`. Otherwise you may overwrite existing stored history, which isn't desirable.

And use `window.history` instead of `history` (not really needed, but let's do it for consistency with `window.history` and `window.location` here/elsewhere in the codebase).